File: ZkTony-WEB/src/main/java/com/zktony/web/entity/LogRecord.java
package com.zktony.web.entity;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;

import java.io.Serial;
import java.io.Serializable;
import java.util.Date;

@Data
@Entity
@Table(name = "log_record")
public class LogRecord implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    @Id
    @Column(name = "id", nullable = false)
    private String id;

    @Column(name = "program_id")
    private String programId;

    @Column(name = "motor")
    private Integer motor;

    @Column(name = "voltage")
    private Float voltage;

    @Column(name = "time")
    private Float time;

    @Column(name = "model")
    private Integer model;

    @Column(name = "upload")
    private Integer upload;

    @Column(name = "create_time")
    @JsonFormat(pattern="yyyy-MM-dd HH:mm:ss", timezone = "GMT+8", locale = "zh")
    private Date createTime;

}

File: ZkTony-ZM/app/src/main/java/com/zktony/www/common/room/entity/LogRecord.kt
/*
 * Copyright (C) 2020 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.zktony.www.common.room.entity

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.Date
import java.util.UUID

/**
 * Data class that represent the a table in the database.
 */
@Entity(tableName = "log_record")
data class LogRecord(
    @PrimaryKey
    var id: String = UUID.randomUUID().toString()
) {
    constructor() : this(UUID.randomUUID().toString())

    var programId = ""
    var model = 0
    var motor = 0
    var voltage = 0f
    var time = 0f
    var upload = 0
    var createTime: Date = Date(System.currentTimeMillis())
}